libs/langchain/langchain/chat_models/base.py

```diff
@@ -200,3 +342,124 @@ def _check_pkg(pkg: str) -> None:
             f"Unable to import {pkg_kebab}. Please install with "
             f"`pip install -U {pkg_kebab}`"
         )
+
+
+def _remove_prefix(s: str, prefix: str) -> str:
+    if s.startswith(prefix):
+        s = s[len(prefix) :]
+    return s
+
+
+_CHAT_MODEL_METHODS_TAKE_CONFIG = tuple(
+    name
+    for name, func in inspect.getmembers(BaseChatModel, inspect.isfunction)
+    if "config" in inspect.signature(func).parameters
+)
+
+_DECLARATIVE_METHODS = (
```

Comment:
(nit) is there a (straightforward) way to test this collection or generate it programmatically, as you do for chat model methods?

Reply:
initially tried to but couldn't find a good programmatic way. Runnable has lots of methods that don't take a config and aren't declarative